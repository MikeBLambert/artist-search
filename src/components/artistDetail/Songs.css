li {
  font-size: 30px;
  color: white;
  text-align: center;
  font-family: 'Roboto', sans-serif;
  margin-top: 15px;


  a {
    color: black;
    text-decoration: none;
    transition: 0.5s;
    &:hover {
      color: #666666;
      font-size: 40px;
      transition: 0.5s;
    }
  }
}

ul {
  list-style-type: none;
  padding: 0;
}