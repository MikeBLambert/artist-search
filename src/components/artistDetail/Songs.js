import React from 'react';
import PropTypes from 'prop-types';

const Songs = ({ songs }) => {
  const songList = songs.map(song => {
    return <li key={song.id}>{song.title}</li>;
  });
  return  <ul>{songList}</ul>;
};

export default Songs;
