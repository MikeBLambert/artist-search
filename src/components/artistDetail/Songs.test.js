import React from 'react';
import Songs from './Songs';
import { shallow } from 'enzyme';

describe('Songs component', () => {

  it('renders a Songs component', () => {
    const nirvanaSongs = [
      { id: 1, name: 'Smells Like Teen Spirit' },
      { id: 2, name: 'Come As You Are' },
      { id: 3, name: 'Lithium' }
    ];
    const wrapper = shallow(<Songs songs={nirvanaSongs}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('displays an artist\'s name and works', () => {
    const nirvanaSongs = [
      { id: 1, name: 'Smells Like Teen Spirit' },
      { id: 2, name: 'Come As You Are' },
      { id: 3, name: 'Lithium' }
    ];
    const wrapper = shallow(<Songs songs={nirvanaSongs}  />);
    expect(wrapper.html()).toContain('Come As You Are');
  })

})