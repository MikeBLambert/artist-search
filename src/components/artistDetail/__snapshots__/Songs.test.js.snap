// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs component renders a Songs component 1`] = `
<Fragment>
  <ul>
    <li
      id="songItem"
      key="1"
    >
      <Link
        replace={false}
        to="/lyrics/Nirvana/undefined"
      />
    </li>
    <li
      id="songItem"
      key="2"
    >
      <Link
        replace={false}
        to="/lyrics/Nirvana/undefined"
      />
    </li>
    <li
      id="songItem"
      key="3"
    >
      <Link
        replace={false}
        to="/lyrics/Nirvana/undefined"
      />
    </li>
  </ul>
</Fragment>
`;
